src/main/java/org/rumbledb/compiler/TranslationVisitor.java

```diff
@@ -445,15 +443,15 @@ public Void visitOrderByExpr(JsoniqParser.OrderByExprContext ctx) {
     @Override
     public Void visitGroupByVar(JsoniqParser.GroupByVarContext ctx) {
         VariableReferenceExpression var;
-        FlworVarSequenceType seq = null;
+        SequenceType seq = null;
         Expression expr = null;
         String uri = null;
         this.visitVarRef(ctx.var_ref);
         var = (VariableReferenceExpression) this.currentExpression;
 
         if (ctx.seq != null) {
             this.visitSequenceType(ctx.seq);
-            seq = (FlworVarSequenceType) this.currentExpression;
+            seq = this.currentSequenceType;
         }
```

Comment:
else condition missing, sequenceType will be left as null

Reply:
DONE